[StructuralMechanics] some compiler warnings

I would have fixed them myself but I don't know how
related to #2612 
~~~
/home/philippb/software/Kratos_Master/applications/StructuralMechanicsApplication/custom_processes/spr_error_process.cpp:275:61: warning: attribute ignored in explicit instantiation ‘class Kratos::SPRErrorProcess<2ul>’ [-Wattributes]
 template KRATOS_API(STRUCTURAL_MECHANICS_APPLICATION) class SPRErrorProcess<2>;
                                                             ^
/home/philippb/software/Kratos_Master/applications/StructuralMechanicsApplication/custom_processes/spr_error_process.cpp:275:61: note: no attribute can be applied to an explicit instantiation
/home/philippb/software/Kratos_Master/applications/StructuralMechanicsApplication/custom_processes/spr_error_process.cpp:276:61: warning: attribute ignored in explicit instantiation ‘class Kratos::SPRErrorProcess<3ul>’ [-Wattributes]
 template KRATOS_API(STRUCTURAL_MECHANICS_APPLICATION) class SPRErrorProcess<3>;
                                                             ^
/home/philippb/software/Kratos_Master/applications/StructuralMechanicsApplication/custom_processes/spr_error_process.cpp:276:61: note: no attribute can be applied to an explicit instantiation
~~~

and

~~~
/home/philippb/software/Kratos_Master/applications/StructuralMechanicsApplication/custom_processes/spr_error_process.cpp: In instantiation of ‘void Kratos::SPRErrorProcess<TDim>::CalculateSuperconvergentStresses() [with long unsigned int TDim = 2ul]’:
/home/philippb/software/Kratos_Master/applications/StructuralMechanicsApplication/custom_processes/spr_error_process.cpp:275:61:   required from here
/home/philippb/software/Kratos_Master/applications/StructuralMechanicsApplication/custom_processes/spr_error_process.cpp:119:60: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
                 for(IndexType i_node_loop = 0; i_node_loop < num_nodes; ++i_node_loop) { // FIXME: Avoid this double loop, extreamily expensive
                                                            ^
/home/philippb/software/Kratos_Master/applications/StructuralMechanicsApplication/custom_processes/spr_error_process.cpp: In instantiation of ‘void Kratos::SPRErrorProcess<TDim>::CalculateSuperconvergentStresses() [with long unsigned int TDim = 3ul]’:
/home/philippb/software/Kratos_Master/applications/StructuralMechanicsApplication/custom_processes/spr_error_process.cpp:276:61:   required from here
/home/philippb/software/Kratos_Master/applications/StructuralMechanicsApplication/custom_processes/spr_error_process.cpp:119:60: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
~~~

--- a/applications/StructuralMechanicsApplication/custom_processes/spr_error_process.cpp
+++ b/applications/StructuralMechanicsApplication/custom_processes/spr_error_process.cpp
@@ -116,7 +116,7 @@ void SPRErrorProcess<TDim>::CalculateSuperconvergentStresses()
                 Vector sigma_recovered_i(SigmaSize,0);
 
                 IndexType count_i = 0;
-                for(IndexType i_node_loop = 0; i_node_loop < num_nodes; ++i_node_loop) { // FIXME: Avoid this double loop, extreamily expensive
+                for(int i_node_loop = 0; i_node_loop < num_nodes; ++i_node_loop) { // FIXME: Avoid this double loop, extreamily expensive
                     auto it_node_loop = nodes_array.begin() + i_node_loop;
                     const SizeType size_elem_neigh = it_node_loop->GetValue(NEIGHBOUR_ELEMENTS).size();
                     if (it_node_loop->Id() == it_neighbour_nodes->Id() && size_elem_neigh > TDim){

--- a/applications/StructuralMechanicsApplication/custom_processes/spr_error_process.cpp
+++ b/applications/StructuralMechanicsApplication/custom_processes/spr_error_process.cpp
@@ -272,7 +272,7 @@ void SPRErrorProcess<TDim>::CalculatePatch(
 /***********************************************************************************/
 /***********************************************************************************/
 
-template KRATOS_API(STRUCTURAL_MECHANICS_APPLICATION) class SPRErrorProcess<2>;
-template KRATOS_API(STRUCTURAL_MECHANICS_APPLICATION) class SPRErrorProcess<3>;
+template class KRATOS_API(STRUCTURAL_MECHANICS_APPLICATION) SPRErrorProcess<2>;
+template class KRATOS_API(STRUCTURAL_MECHANICS_APPLICATION) SPRErrorProcess<3>;
 
 };// namespace Kratos.

